router/table.go

```diff
@@ -19,15 +19,18 @@ var (
 // table is an in-memory routing table
 type table struct {
 	sync.RWMutex
+	// router which manages the table
+	router *router
```

Comment:
Not a fan of this circular dependency. Any way we can break this ?

Reply:
I can either:
• Pass the fetchRoutes function as an argument when creating the table, it's still a kind of circular dependancy but less direct.
• Merge the table struct and the router struct. Right now there the table is "dumb" and has no knowledge of the registry or where any of its data comes from.

Reply:
Let's pass in `fetchRoutes`. Thanks